pkg/agent/metrics/collector.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"strconv"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/ovs/ovsctl"
+)
+
+type ovsMeterStatsCollector struct {
+	ovsCtlClient ovsctl.OVSCtlClient
+}
+
+func NewOVSMeterStatsCollector(ovsCtlClient ovsctl.OVSCtlClient) *ovsMeterStatsCollector {
+	return &ovsMeterStatsCollector{ovsCtlClient}
+}
+
+func (collector *ovsMeterStatsCollector) Run(stopCh <-chan struct{}) {
+	wait.Until(collector.collectMeterStats, time.Second*30, stopCh)
+}
+
+func (collector *ovsMeterStatsCollector) collectMeterStats() {
+	klog.InfoS("Collect meter stats")
+	statsMap, err := collector.ovsCtlClient.GetMeterStats()
+	if err != nil {
+		klog.InfoS("Failed to get meter stats", "error msg", err)
+	}
+	for key, val := range statsMap {
+		klog.InfoS("key value pair", "key", key, "val", val)
+		OVSMeterPacketCount.WithLabelValues(strconv.Itoa(key)).Set(float64(val))
```

Comment:
Is it possible to report both packet out and byte out in one meter?
Can we add a text label to explain "meter 1" and "meter 2" mean? Otherwise they are hard to understand for both developers and users.

Reply:
I think there should be only one value for one meter.
To report byte count,  we can add another byte_count label for this OVSMeterPacketCount.
For text label,  I am thinking to replace current meter_id (1 and 2) by PacketInMeterIDNP/PacketInMeterIDTF.
What do you think?

Reply:
Forget my last comment regarding add byte_count as label. It will be a different meter in that case. I cannot think of a way to report byte count in one meter. In my view, new meter OVSMeterByteCount is needed for that.

Reply:
@tnqn I have changed the meter_id to PacketInMeterIDNP/PacketInMeterIDTF instead. Please help check the latest version. I don't think we can report both both packet count and byte count in one meter. Do you think we need to add a new one for byte count?

Reply:
No need to add byte counter in this PR, we can see if it's needed in the future.